src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -737,32 +739,55 @@ private static bool IsTypeLessVisibleThan(TypeSymbol type, Symbol sym, ref Compo
                     case TypeKind.Struct:
                     case TypeKind.Interface:
                     case TypeKind.Delegate:
-                        var typeArguments = ((NamedTypeSymbol)current).TypeArgumentsWithAnnotationsNoUseSiteDiagnostics;
-                        if (typeArguments.IsEmpty)
+
+                        if (current.IsAnonymousType && current.TypeKind == TypeKind.Class)
                         {
-                            return null;
-                        }
+                            var anonymousClass = (AnonymousTypeManager.AnonymousTypePublicSymbol)current;
+                            var properties = anonymousClass.Properties;
+
+                            if (properties.IsEmpty)
+                            {
+                                return null;
+                            }
+
+                            int i;
+                            for (i = 0; i < properties.Length - 1; i++)
+                            {
+                                if (visitTypeWithAnnotations(
+                                        properties[i].TypeWithAnnotations,
+                                        typeWithAnnotationsPredicate, typePredicate, arg,
+                                        useDefaultType, canDigThroughNullable, visitCustomModifiers) is TypeSymbol result)
+                                {
+                                    return result;
+                                }
+                            }
 
-                        int i;
-                        for (i = 0; i < typeArguments.Length - 1; i++)
+                            next = properties[i].TypeWithAnnotations;
+                        }
+                        else
                         {
-                            // Let's try to avoid early resolution of nullable types
-                            (TypeWithAnnotations nextTypeWithAnnotations, TypeSymbol? nextType) = getNextIterationElements(typeArguments[i], canDigThroughNullable);
-                            var result = VisitType(
```

Comment:
>VisitType [](http://example.com/codeflow?start=41&length=9)

I do not think the change is necessary, we just need to propagate ```visitCustomModifiers``` and ```VisitType``` should do the right thing.